cogs/helpermessage.py

```diff
@@ -0,0 +1,144 @@
+        self.bot = bot
+        self.data = toml.load('config.toml')
+        self.subjects = self.data['subjects']
+        self.helpermessages = self.data['helpermessages']
+        self.subject_channels = self.subjects.keys()
+        self.helper_roles = [self.subjects[channel] for channel in self.subject_channels]
+
+    @commands.hybrid_group()
+    async def helpermessage(self, ctx: commands.Context):
+        """
+        Commands for handling reminders.
+        """
+        await available_subcommands(ctx)
+
+    @helpermessage.command()
+    async def list(self, ctx: commands.Context):
+        """
+        List all active helpermessage embeds
+        """
+        embed_dict = dict(
+            title = 'List of all active helpermessages',
+            description = f'Here is a list of all active helpermessages.',
+        )
+        names = [f'<#{channel}>' for channel in self.helpermessages.keys()]
+        values = [f'https://discord.com/channels/{ctx.guild.id}/{channel}/{int(self.helpermessages[channel])}' for channel in self.helpermessages.keys()]
+        embeds = paginated_embed_menus(names, values, embed_dict=embed_dict)
+        embed, view = await PaginationView(ctx, embeds).return_paginated_embed_view()
+        await ctx.send(embed=embed, view=view)
+
+    async def embed_getter(self, edited_roles):
+        for role in edited_roles:
+            for channel, roles in self.subjects.items():
+                if isinstance(role, list):
+                    if role.id in roles:
+                        discord_channel = await self.bot.fetch_channel(int(channel))
+                        helpermessage = await discord_channel.fetch_message(int(self.helpermessages[channel]))
+                        embed = helpermessage.embeds[0]
+                        return helpermessage, role, embed
+                elif role.id == roles:
+                        discord_channel = await self.bot.fetch_channel(int(channel))
+                        helpermessage = await discord_channel.fetch_message(int(self.helpermessages[channel]))
+                        embed = helpermessage.embeds[0]
+                        return helpermessage, role, embed
+        return None, None, None
+
+    @commands.Cog.listener()
+    async def on_member_update(self, before: discord.Member, after: discord.Member):
+        """
+        Update helper message based on user helper/dehelper.
+        """
+        # checking if there has been a change in roles
+        if before.roles != after.roles: 
+            before_roles = set(before.roles)
+            after_roles = set(after.roles)
+            # checking whether roles were removed or added
+            added_roles = after_roles - before_roles
+            removed_roles = before_roles - after_roles
+            # routine that runs to add the role
+            if added_roles:
+                helpermessage, role, embed = await self.embed_getter(added_roles)
```

Comment:
Because of the bugs I have annotated in the `embed_getter` method, maybe it's worth turning this into a generator that `yield`s these values rather than `return`ing them. You would then change this into a `for` loop, and perform the addition/removal of the user inside the loop.